[bug] CONAN_CMAKE_PROGRAM not respected for CMake Toolchain

### Environment Details (include every applicable attribute)
  * Operating System+version:Linux/Windows
  * Compiler+version:gcc/MSVC
  * Conan version:HEAD
  * Python version:3.8

### Steps to reproduce (Include if Applicable)
Have CMake 2.8 installed on your system but also have CMake 3.24 installed to another location.
Specify the CMake version required within CONAN_CMAKE_PROGRAM environment variable.
Try to compile zlib from the conan center recipie. Shouldn't matter the version they have all been converted to use CMake toolchain and specify that CMake minimum version of >2.8.

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.0 or higher is required.  You are running version 2.8.12.2


-- Configuring incomplete, errors occurred!
```

### Source of issue
https://github.com/conan-io/conan/pull/4298 added this feature back in 2019 but when the CMake Toolchain refactor was done in 2020 https://github.com/conan-io/conan/pull/7076 the #4298 change to cmake program were not ported over. 

Problem line is:
https://github.com/conan-io/conan/blame/develop/conan/tools/cmake/cmake.py#L62

Just needs to be updated to follow 4298:
`self._cmake_program = os.getenv("CONAN_CMAKE_PROGRAM") or cmake_program or "cmake"`

I've locally applied this and it fixed my issue.

Of course its completely possible that I am misunderstanding how to set the cmake program path.

--- a/conan/tools/cmake/cmake.py
+++ b/conan/tools/cmake/cmake.py
@@ -49,7 +49,7 @@ class CMake(object):
         self._toolchain_file = configure_preset.get("toolchainFile")
         self._cache_variables = configure_preset["cacheVariables"]
 
-        self._cmake_program = "cmake"  # Path to CMake should be handled by environment
+        self._cmake_program = conanfile.conf.get("tools.cmake:cmake_program", default="cmake")
 
     def configure(self, variables=None, build_script_folder=None, cli_args=None):
         """

--- a/conans/model/conf.py
+++ b/conans/model/conf.py
@@ -63,6 +63,7 @@ BUILT_IN_CONFS = {
     "tools.cmake.cmaketoolchain:system_processor": "Define CMAKE_SYSTEM_PROCESSOR in CMakeToolchain",
     "tools.cmake.cmaketoolchain:toolset_arch": "Toolset architecture to be used as part of CMAKE_GENERATOR_TOOLSET in CMakeToolchain",
     "tools.cmake.cmake_layout:build_folder_vars": "Settings and Options that will produce a different build folder and different CMake presets names",
+    "tools.cmake:cmake_program": "Path to CMake executable",
     "tools.files.download:retry": "Number of retries in case of failure when downloading",
     "tools.files.download:retry_wait": "Seconds to wait between download attempts",
     "tools.gnu:make_program": "Indicate path to make program",

--- a/conans/test/unittests/tools/cmake/test_cmake_presets_definitions.py
+++ b/conans/test/unittests/tools/cmake/test_cmake_presets_definitions.py
@@ -30,6 +30,17 @@ def conanfile():
     return c
 
 
+def test_cmake_cmake_program(conanfile):
+    mycmake = "C:\\mycmake.exe"
+    conanfile.conf.define("tools.cmake:cmake_program", mycmake)
+
+    with mock.patch("platform.system", mock.MagicMock(return_value='Windows')):
+        write_cmake_presets(conanfile, "the_toolchain.cmake", "MinGW Makefiles", {})
+
+    cmake = CMake(conanfile)
+    assert cmake._cmake_program == mycmake
+
+
 def test_cmake_make_program(conanfile):
     def run(command):
         assert '-DCMAKE_MAKE_PROGRAM="C:/mymake.exe"' in command
@@ -42,4 +53,3 @@ def test_cmake_make_program(conanfile):
 
     cmake = CMake(conanfile)
     cmake.configure()
-

